docs/docs/build/building-apps/05-protobuf-annotations.md

```diff
@@ -0,0 +1,81 @@
+---
+sidebar_position: 1
+---
+
+# ProtocolBuffer Annotations
+
+This document explains the various protobuf scalars that have been added to make working with protobuf easier for Cosmos SDK application developers
+
+## Signer
+
+Signer specifies which field should be used to determine the signer of a message for the Cosmos SDK. This field can be used for clients as well to infer which field should be used to determine the signer of a message.
+
+Read more about the signer field [here](../building-modules/02-messages-and-queries.md).
+
+```protobuf reference 
+https://github.com/cosmos/cosmos-sdk/blob/e6848d99b55a65d014375b295bdd7f9641aac95e/proto/cosmos/bank/v1beta1/tx.proto#L40
+```
+
+```proto
+option (cosmos.msg.v1.signer) = "from_address";
+```
+
+## Scalar
+
+The scalar type defines a way for clients to understand how to construct protobuf messages according to what is expected by the module and sdk.
+
+```proto
+(cosmos_proto.scalar) = "cosmos.AddressString"
+```
+
+There are a few options for what can be provided as a scalar: cosmos.AddressString, cosmos.ValidatorAddressString, cosmos.ConsensusAddressString, cosmos.Int, cosmos.Dec. 
```

Comment:
Can we maybe give an example? It is quite important that the scalar annotations are correct for the addresses as AutoCLI relies on it (otherwise it will just be considered as a string without any validation)

Reply:
i think you should assume it wont be correct and/or they wont be implemented at all. I can add it but we shouldnt assume its used

Reply:
I feel like it is the role of developer removing their CLI for AutoCLI to verify the annotation. Hence I thought a small mention was useful.

However, AutoCLI will still work without it. Addresses will simply not be verified (if you have an annotation there will be a field verification that you have a valid address).

Reply:
for sure, added it. It will be hard to get people to adopt this. Many times with proto its write once and forget till you need a change. Adding annotations will be harder to for people to integrate